src/System.Buffers/src/System/Buffers/IBufferPool.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace System.Buffers
+{
+    public interface IBufferPool<T>
```

Comment:
I originally made the abstraction for the ASP.NET case where they look to need byte[] and char[] buffers, but I think I can refactor this PR a little bit more to make it flexible for both without the abstraction